File: 11기 코딩테스트_230308/연습2/PrimeNumber/PrimeNumber_test.py
import unittest

from src.com.luckydave.algorithm.fastcampus.supereasy.PrimeNumber.PrimeNumber import solution


class CustomTestCase(unittest.TestCase):

    def test_1(self):
        # given
        n = 7

        # when
        result = solution(n)

        # then
        self.assertEqual(True, result)

    def test_2(self):
        # given
        n = 417

        # when
        result = solution(n)

        # then
        self.assertEqual(False, result)

    def test_3(self):
        # given
        n = 633

        # when
        result = solution(n)

        # then
        self.assertEqual(False, result)

    def test_4(self):
        # given
        n = 92837

        # when
        result = solution(n)

        # then
        self.assertEqual(False, result)

    def test_5(self):
        # given
        n = 982374

        # when
        result = solution(n)

        # then
        self.assertEqual(False, result)

    def test_6(self):
        # given
        n = 73049183

        # when
        result = solution(n)

        # then
        self.assertEqual(True, result)


if __name__ == '__main__':
    unittest.main()

File: 11기 코딩테스트_230308/연습2/PrimeNumber/PrimeNumber_test.js
import assert from 'assert'
import solution from './PrimeNumber.js'

describe('Custom Test', function () {
  it('test 1', function (done) {
    // given
    const n = 7

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });

  it('test 2', function (done) {
    // given
    const n = 417

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 3', function (done) {
    // given
    const n = 633

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 4', function (done) {
    // given
    const n = 92837

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 5', function (done) {
    // given
    const n = 982374

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(false, result)
    done();
  });

  it('test 6', function (done) {
    // given
    const n = 73049183

    // when
    const result = solution(n)

    // then
    assert.deepStrictEqual(true, result)
    done();
  });
})
